src/views/document/utils/TagsButton.vue

```diff
@@ -0,0 +1,64 @@
+<template>
+  <a v-if="canTag" @click="onClick" :title="tooltip" :class="{ 'todo-button-yellow': todo }">
+    <fa-icon icon="star" />
+  </a>
+</template>
+
+<script>
+import c2c from '@/js/apis/c2c';
+import { requireDocumentProperty } from '@/js/properties-mixins';
+
+export default {
+  mixins: [requireDocumentProperty],
+
+  data() {
+    return {
+      todo: null,
+    };
+  },
+
+  computed: {
+    canTag() {
+      return this.document.type === 'r' && this.$user.isLogged;
+    },
+
+    tooltip() {
+      if (this.todo) {
+        return this.$gettext('Unmark this route as to do');
+      } else {
+        return this.$gettext('Mark this route as to do');
+      }
+    },
+  },
+
+  created() {
+    if (this.canTag) {
+      c2c.tags.get(this.document).then((response) => {
```

Comment:
you don't handle API errors. OK to fail quite silently, but maybe we should disable the star icon and add a error badge on it in this case, or something like that?